### PR TITLE
[Dy2St] enable_to_static_guard 推全 40-45

### PR DESCRIPTION
**Pull request by ooooo-create**
### PR types

Others 
### PR changes

Others 
### Description

https://github.com/PaddlePaddle/Paddle/issues/59684

@DrRyanHuang @SigureMo

### CODE REVIEW

**Review by DrRyanHuang — APPROVED** [review]
LGTM🔍

File: test/dygraph_to_static/test_tensor_methods.py [hunk 1]
```diff
@@ -118,16 +123,17 @@ def true_div(x, y):
...
         y = paddle.to_tensor([4], dtype='int64')
         return paddle.jit.to_static(true_div)(x, y).numpy()
 
     @test_legacy_and_pt_and_pir
     def test_ture_div(self):
         paddle.disable_static()
-        dygraph_res = self._run(to_static=False)
+        with enable_to_static_guard(False):
+            dygraph_res = self._run(to_static=False)
+
```
Comment (DrRyanHuang) [inline test/dygraph_to_static/test_tensor_methods.py]:
```suggestion
        with enable_to_static_guard(False):
            dygraph_res = self._run()
```

**Review by DrRyanHuang — CHANGES_REQUESTED** [review]

File: test/dygraph_to_static/test_tensor_methods.py [hunk 2]
```diff
@@ -118,16 +123,17 @@ def true_div(x, y):
...
 class TestTrueDiv(Dy2StTestBase):
-    def _run(self, to_static):
-        paddle.jit.enable_to_static(to_static)
+    def _run(self):
         x = paddle.to_tensor([3], dtype='int64')
         y = paddle.to_tensor([4], dtype='int64')
         return paddle.jit.to_static(true_div)(x, y).numpy()
 
     @test_legacy_and_pt_and_pir
     def test_ture_div(self):
```
Comment (DrRyanHuang) [inline test/dygraph_to_static/test_tensor_methods.py]: 顺便也改一下吧
```suggestion
    def test_true_div(self):
```

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**